unittests/linuxliketests.py

```diff
@@ -1411,25 +1411,26 @@ def install_subdir_invalid_symlinks(self, testdir, subdir_path):
         Test that installation of broken symlinks works fine.
         https://github.com/mesonbuild/meson/issues/3914
         '''
-        testdir = os.path.join(self.common_test_dir, testdir)
-        subdir = os.path.join(testdir, subdir_path)
-        with chdir(subdir):
-            # Can't distribute broken symlinks in the source tree because it breaks
-            # the creation of zipapps. Create it dynamically and run the test by
-            # hand.
-            src = '../../nonexistent.txt'
-            os.symlink(src, 'invalid-symlink.txt')
-            try:
-                self.init(testdir)
-                self.build()
-                self.install()
-                install_path = subdir_path.split(os.path.sep)[-1]
-                link = os.path.join(self.installdir, 'usr', 'share', install_path, 'invalid-symlink.txt')
-                self.assertTrue(os.path.islink(link), msg=link)
-                self.assertEqual(src, os.readlink(link))
-                self.assertFalse(os.path.isfile(link), msg=link)
-            finally:
-                os.remove(os.path.join(subdir, 'invalid-symlink.txt'))
+        # We can't distribute broken symlinks in the source tree because it
+        # breaks the creation of zipapps. Create the symlink dynamically and run
+        # the test by hand. Do this in a tmpdir in order to not tread on
+        # parallel tests' toes.
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
I think what we usually do in such cases is use self.builddir as the tmpdir, then use `self.new_builddir()` to get a new one for actual building.

Reply:
Do you have an example of that getting used this way?

Reply:
Could I perhaps interest you in a dedicated helper? https://github.com/mesonbuild/meson/pull/12539/commits/fc33db22d7a987facc61f0d40a13dc6924b80c33

Reply:
Seems good. Thanks!